### ISSUE TITLE
Inconsistent behavior between ByteTensor and Variable(ByteTensor): bug or feature?

### ISSUE THREAD

**Issue by WarBean** [pytorch/pytorch#1389]
Test script:

```
import torch
from torch.autograd import Variable

byte_tensor = torch.ByteTensor(256)
byte_tensor[:] = 1
byte_variable = Variable(byte_tensor)
# The results should be the same, but actually not
print('byte_tensor.sum():')
print(byte_tensor.sum())
print('byte_variable.sum():')
print(byte_variable.sum())
```

Results:

```
byte_tensor.sum():
256
byte_variable.sum():
Variable containing:
 0
[torch.ByteTensor of size 1]
```

Looks like `ByteTensor` **does handle overflow** but `Variable(ByteTensor)` **doesn't**. Such inconsistency has caused some troubles when I'm calculating my model's accuracy.

**Comment by soumith** [comment]
This happens because Variables always return a 1-element Tensor for functions that return scalars.
And when we try to construct a 1-element ByteTensor with the value 256, it overflows to 0.

This is expected until we introduce a scalar type into autograd 
( https://github.com/pytorch/pytorch/issues/1433 )

**Comment by gchanan** [comment]
this has been fixed.
